File: 1350-remove-sub-folders-from-the-filesystem/1350-remove-sub-folders-from-the-filesystem.cpp
class Solution {
public:
    vector<string> removeSubfolders(vector<string>& folder) {
        unordered_set<string> st(folder.begin(),folder.end());

        for(int i=0;i<folder.size();i++){
            string subFolder=folder[i];
            string str=subFolder;
            while(!str.empty()){
                auto pos=str.find_last_of('/'); //returns the last position of the char in a string
                str=subFolder.substr(0,pos);
                if(st.find(str)!=st.end()){
                    st.erase(subFolder); //already present
                    break;
                }
            }
        }

        vector<string>ans(st.begin(),st.end());
        return ans;
    }
};

File: 1350-remove-sub-folders-from-the-filesystem/1350-remove-sub-folders-from-the-filesystem.java
class Solution {
    public List<String> removeSubfolders(String[] folder) {
        Set<String> st= new HashSet<>(Arrays.asList(folder));

        for(String str:folder){
            String subFolder=str;
            while(!subFolder.isEmpty()){
                int pos=subFolder.lastIndexOf('/');
                subFolder=subFolder.substring(0,pos);
                if(st.contains(subFolder)){
                    st.remove(str);
                    break;
                }
            }
        }
        return new ArrayList<>(st);
    }
}